File: Source/Run/SPA/src/app/models/csv-import-result.ts
export interface CsvImportRowError {
  lineNumber: number;
  field?: string;
  message: string;
}

export interface CsvImportResult {
  rowsTotal: number;
  created: number;
  updated: number;
  skipped: number;
  failed: number;
  rowErrors: CsvImportRowError[];
}

File: Source/_Core/Contracts/Api/Responses/CsvImportResult.cs
namespace WealthLedger.Contracts.Api.Responses;

public class CsvImportResult
{
    public int RowsTotal { get; set; }
    public int Created { get; set; }
    public int Updated { get; set; }
    public int Skipped { get; set; }
    public int Failed { get; set; }
    public List<CsvImportRowError> RowErrors { get; set; } = new();
}
